vars/dockerExecute.groovy

```diff
@@ -1,14 +1,31 @@
 import com.cloudbees.groovy.cps.NonCPS
 
 def call(Map parameters = [:], body) {
+
+    def PLUGIN_ID_DOCKER_WORKFLOW = 'docker-workflow'
+
     handlePipelineStepErrors(stepName: 'dockerExecute', stepParameters: parameters){
         def dockerImage = parameters.get('dockerImage', '')
         Map dockerEnvVars = parameters.get('dockerEnvVars', [:])
         def dockerOptions = parameters.get('dockerOptions', '')
         Map dockerVolumeBind = parameters.get('dockerVolumeBind', [:])
 
-        if(dockerImage?.isEmpty()){
-            echo '[dockerExecute] No Docker image provided - running on local environment.'
+        if(dockerImage) {
+
+            if (! Jenkins.instance.pluginManager.plugins.find { p -> p.isActive() && p.getShortName() == PLUGIN_ID_DOCKER_WORKFLOW } ) {
```

Comment:
Would be interesting to get comments on that approach. Maybe we should make that to a common pattern. In this case there should be a corresponding method in some auxiliary class.
We have a lot of implicit dependencies to Jenkins plugins. This would be a way to make these dependencies transparent. There is not much benefit with this approch in case we run inside some docker Jenkins where we can ensure availability of plugins. But such an approach in beneficial in case we cannot make strong assumptions about the installed plugins.

Reply:
I would avoid checking for single plugins in all cases.
The error message in Jenkins is pretty speaking if a plugin is not available. It will tell that a specific command could not be found.

Reply:
But it helps make plugin dependencies transparent. Plugin dependencies are not declared good on the level of the library. They are defined by some plugin lists outside the library.